scripts/change_did_key.js

```diff
@@ -0,0 +1,113 @@
+import { Keyring } from '@polkadot/api';
+import { cryptoWaitReady } from '@polkadot/util-crypto';
+import { u8aToHex } from '@polkadot/util';
+import { createSignedKeyUpdate } from '../src/utils/did';
+import { getPublicKeyFromKeyringPair } from '../src/utils/misc';
+import { DockAPI, PublicKeySr25519 } from '../src/api';
+import b58 from 'bs58';
+
+/// This script is not configurable via command line. You will need to maually edit `main()`
+/// to input your own values.
+///
+/// Use (with node running locally):
+///
+/// ```bash
+/// cd sdk
+/// yarn
+/// npx babel-node ./scripts/change_did_key.js
+/// ```
+
+(async () => {
+  await main();
+  process.exit(0);
+})().catch(e => {
+  console.error(e);
+  process.exit(1);
+});
+
+async function main() {
+  /// This did is preregistered for local development chains.
+  const ALICE_DID = '0x416c696365000000000000000000000000000000000000000000000000000000';
+  /// The default keypair for "Alice\0\0\0..." on local developnent chains.
+  const ALICE_KP = {
+    sk: '0x416c696365736b00000000000000000000000000000000000000000000000000',
+    pk: '0xb8b868f83227df83240fdbefcaa2636bebf16304c9249069897b510b7d8bbd0b',
+  };
+  /// Some arbitray sr25519 keypair.
+  const TEST_KP = {
+    sk: '0x6b9abb2153e0e3989bb53869b4a88806941a0a86cefa08a7977aff73379df734',
+    pk: '0xc081a33d27d66ceda4e9f634f2f3a1b3b7f4ad898662036416250862ffae623f',
+  };
+
+  // set key to TEST_KP
+  await updateDIDKey({
+    nodeWsUrl: 'ws://localhost:9944',
+    did: ALICE_DID,
+    currentSk: ALICE_KP.sk,
+    newPk: TEST_KP.pk,
+    payerSk: "//Bob"
+  });
+
+  // set it back to the default
+  await updateDIDKey({
```

Comment:
Why is the update reverted?

Reply:
For testing. It makes it so you can run this script repeatedly without needing to `purge-chain`.

Reply:
But that would require the actual user to comment out this before he runs the script?

Reply:
This function call is no longer present.